### PR TITLE
Refactor startcontroller

### PR DESCRIPTION
**Pull request by doobry-systemli**
Refactor StartController and twig template `Start/index.html.twig`

### CODE REVIEW

File: config/routes.yaml [hunk 1]
```diff
@@ -1,3 +1,25 @@
+# StartController
+
+index:
+    path:     /
+    controller: App\Controller\StartController::indexAction
+
+password_change:
+    path:     /password_change
```
Comment (0x46616c6b) [inline config/routes.yaml]: I would prefer `/user/password/change`

File: config/routes.yaml [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+    controller: App\Controller\StartController::passwordChangeAction
+
+voucher_create:
+    path:     /voucher_create
```
Comment (0x46616c6b) [inline config/routes.yaml]: I would prefer `/voucher/new`

File: config/routes.yaml [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+    controller: App\Controller\StartController::voucherCreateAction
+
+# DeleteController
+
+delete:
+    path:     /delete
```
Comment (0x46616c6b) [inline config/routes.yaml]: I would prefer `/user/delete`

File: src/Controller/StartController.php [hunk 4]
```diff
@@ -30,79 +30,126 @@ class StartController extends Controller
...
 
-                    if ($form->isSubmitted() && $form->isValid()) {
-                        $user->setPlainPassword($passwordChange->newPassword);
+        $voucherCreateForm = $this->getVoucherCreateForm();
 
-                        $this->get(PasswordUpdater::class)->updatePassword($user);
+        return $this->render('Start/index.html.twig', array(
+            'user' => $user,
+            'vouchers' => $vouchers,
+            'form' => $passwordChangeForm->createView(),
```
Comment (0x46616c6b) [inline src/Controller/StartController.php]: We should rename `form` variable to more speaking `password_form`

File: src/Controller/StartController.php [hunk 5]
```diff
@@ -30,79 +30,126 @@ class StartController extends Controller
...
-                        return $this->redirect($this->generateUrl('index'));
+                    if ($voucher instanceof Voucher) {
+                        $request->getSession()->getFlashBag()->add('success', 'flashes.voucher-creation-successful');
                     }
                 }
+
+                return $this->redirect($this->generateUrl('index'));
             }
+        } else {
+            return $this->redirect($this->generateUrl('index'));
```
Comment (0x46616c6b) [inline src/Controller/StartController.php]: You can add only one return statement at the end.

File: src/Controller/StartController.php [hunk 6]
```diff
@@ -30,79 +30,126 @@ class StartController extends Controller
...
+
+                return $this->redirect($this->generateUrl('index'));
+            }
         } else {
-            return $this->render('Start/index_anonymous.html.twig');
+            return $this->redirect($this->generateUrl('index'));
         }
     }
+
+    private function getVoucherCreateForm()
```
Comment (0x46616c6b) [inline src/Controller/StartController.php]: Missing Doctype

Comment (doobry-systemli) [reply src/Controller/StartController.php]: done

**Pull request closed without merging**